lib/puppet/provider/ec2_securitygroup/v2.rb

```diff
@@ -23,7 +23,7 @@ def self.instances
   def self.prefetch(resources)
     instances.each do |prov|
       if resource = resources[prov.name] # rubocop:disable Lint/AssignmentInCondition
-        resource.provider = prov if resource[:region] == prov.region
+        resource.provider = prov if (resource[:region] || ENV['AWS_REGION']) == prov.region
```

Comment:
This is actually fairly tricky, and I'm not sure that this is the right approach.

Assume that:

* AWS has 2 security groups named `foo` in `us-east-1` and `us-west-1`.  
* In a manifest, you've defined a sg resource with `:region => 'us-west-1'`
* You've set `AWS_REGION=us-east-1`

It's now possible that even though the manifest wants to modify the resource in `us-west-1`, that Puppet will attach to and modify the resource in `us-east-1`

This may seem like an unlikely / synthesized problem, but it makes me really wary of doing something like this.  I'm not sure there's a reasonable way around this problem.